### DISCUSSION TITLE
Draft files

### DISCUSSION THREAD

**Discussion by eliasboegel** [quarto-dev/quarto-cli#4246] [Feature Requests]
Next to quarto, I've used [mdbook](https://github.com/rust-lang/mdBook) for some documentation. One thing I found very helpful was the feature they call "draft chapters". Essentially, it allows you to add a mock file to the included file list to have it show up as a grayed out section in the index and TOC. This would primarily be used when you the document is still WIP, but you want to make the intended structure and future added content clear to any readers of the WIP version. I see an application first and foremost for the "book" type. 

The mdbook guide features an example: https://rust-lang.github.io/mdBook/format/summary.html
![image](https://user-images.githubusercontent.com/62483075/217113399-f1740e34-f795-49a6-85b9-c591ac9efbba.png)

Shows up as:
![image](https://user-images.githubusercontent.com/62483075/217113471-18d8fd02-32cd-46c3-804f-cae6ac195db8.png)

I would love to have this to block out a full structure of, for example, a book. In mdbook, a draft chapter is added by adding a chapter with name but not file. In Quarto, it seems like no file is not an option. The best way I see to implement this is to introduce some sort of marker in the file front matter that can be used to mark a file as a draft, where the file should be included in the document structure, but any links to the page should be inactive and not clickable and the page should not be rendered. The same also would translate to other publishing formats like PDF - content isn't present except in the TOC. This is a bit in contrast to draft posts, which (to my understanding) are just skipped entirely without inclusion to the index and TOC. 

The front matter indicator could look like this (similar to draft posts) :
![image](https://user-images.githubusercontent.com/62483075/217115389-3aa75e26-f510-46bf-a7bb-cb9e4251fedf.png)

**Comment by mcanouil** [comment]
For reference, it's related to #2768 (associated opened issue #2781) and in some apect to #3114

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/2781
